### ISSUE TITLE
Allow user to auto-split test set from their training file

### ISSUE THREAD

**Issue by ledell** [h2oai/wave-h2o-automl#47]
Currently we allow the user to optionally upload a test file, but we really always want to have a test set for the Explain functions.

If the user only uploads a training csv, then we should automatically create a test set that will be used in the Explain functions.  We should allow the user to adjust the fraction of their training data that will be used for the test set.

Related to this: https://github.com/h2oai/wave-h2o-automl/issues/14
